tests/e2e/setup/create-cluster.sh

```diff
@@ -0,0 +1,280 @@
+  make deploy
+  echo -e "${GREEN}<= GreptimeDB operator is deployed.${RESET}"
+}
+
+# Deploy etcd cluster that used for metasrv.
+function deploy_etcd_cluster() {
+  echo -e "${GREEN}=> Deploy etcd cluster...${RESET}"
+  helm upgrade --install etcd oci://registry-1.docker.io/bitnamicharts/etcd \
+    --set replicaCount=1 \
+    --set auth.rbac.create=false \
+    --set auth.rbac.token.enabled=false \
+    --namespace "$ETCD_NAMESPACE" \
+    --create-namespace \
+    --version "$ETCD_CHART_VERSION"
+  echo -e "${GREEN}<= etcd cluster is deployed.${RESET}"
+}
+
+# Deploy Kafka cluster using Strimzi for remote WAL testing.
+function deploy_kafka_cluster() {
+  echo -e "${GREEN}=> Deploy Kafka cluster...${RESET}"
+  kubectl create namespace "$KAFKA_NAMESPACE"
+  kubectl create -f "https://strimzi.io/install/latest?namespace=$KAFKA_NAMESPACE" -n "$KAFKA_NAMESPACE"
+  kubectl apply -f ./tests/e2e/setup/kafka-wal.yaml -n "$KAFKA_NAMESPACE"
+  echo -e "${GREEN}<= Kafka cluster is deployed.${RESET}"
+}
+
+# Deploy cloud-provider-kind for using LoadBalancer service type.
+function deploy_cloud_provider_kind() {
+  if ! hash cloud-provider-kind 2>/dev/null; then
+    go install sigs.k8s.io/cloud-provider-kind@latest
+  fi
+
+  echo -e "${GREEN}=> Deploy cloud-provider-kind...${RESET}"
+  kubectl label node "${CLUSTER_NAME}-control-plane" node.kubernetes.io/exclude-from-external-load-balancers-
+
+  # Check if the os is darwin or linux.
+  # If the os is darwin, we need to use sudo to run cloud-provider-kind.
+  nohup cloud-provider-kind > /tmp/cloud-provider-kind-logs 2>&1 &
+  echo -e "${GREEN}<= cloud-provider-kind is deployed.${RESET}"
+}
+
+function check_kafka_cluster_status() {
+  wait_for \
+    "kubectl -n $KAFKA_NAMESPACE get kafka $KAFKA_CLUSTER_NAME -o json | jq '.status.conditions[] | select(.type==\"Ready\" and .status==\"True\")' | grep Ready" \
+    "${DEFAULT_TIMEOUT%s}"
+}
+
+function wait_for() {
+  local command="$1"
+  local timeout="$2"
+  local interval=1
+  local elapsed_time=0
+
+  while true; do
+    # Turn off the exit-on-error flag to avoid exiting the script.
+    set +e
+    eval "$command" &> /dev/null
+    local status=$?
+    set -e
+
+    # Check if command was successful.
+    if [[ $status -eq 0 ]]; then
+      return 0
+    fi
+
+    # Update elapsed time.
+    ((elapsed_time += interval))
+
+    # Check if the timeout has been reached.
+    if [[ $elapsed_time -ge $timeout ]]; then
+      echo "Timeout reached. Command failed."
+      return 1
+    fi
+
+    # Wait for a specified interval before retrying.
+    sleep $interval
+  done
+}
+
+function wait_all_service_ready() {
+  echo -e "${GREEN}=> Wait for all services to be ready...${RESET}"
+
+  # Wait for etcd to be ready.
+  kubectl wait \
+    --for=condition=Ready \
+    pod -l app.kubernetes.io/instance=etcd \
+    -n "$ETCD_NAMESPACE" \
+    --timeout="$DEFAULT_TIMEOUT"
+
+  # Wait for kafka to be ready.
+  check_kafka_cluster_status
+
```

Comment:
suggestion: add wait for greptimedb ready.

Reply:
We don't deploy greptimedb in `create-cluster.sh`.

Reply:
Ignore it.